taskfile/read/taskfile.go

```diff
@@ -141,3 +144,28 @@ func readTaskfile(file string) (*taskfile.Taskfile, error) {
 	var t taskfile.Taskfile
 	return &t, yaml.NewDecoder(f).Decode(&t)
 }
+
+func exists(dir, name string) (string, string, error) {
+	var path string
+	if name != "" {
+		path = filepath.Join(dir, name)
+		if _, err := os.Stat(path); err != nil {
+			return path, name, fmt.Errorf(`task: No Taskfile found on "%s". Use "task --init" to create a new one`, path)
+		}
+		return path, name, nil
```

Comment:
One of the reasons I was hesitant to do this before is because it could generate some bugs on inclusion of external Taskfiles.

Here we probably have one: if the given entry point was `Foobar.yaml` we don't want for included directories to look for `dir/Foobar.yaml`, but for `dir/Taskfile.yaml`. We need to remember the extension for inclusion, but not the actual file name.

Reply:
I played around with this a bit and wrote some tests and tbh I'm not sure which is a better behaviour. You are correct, if your entrypoint is foobar.yaml then includes without filename results in an error:

`open included/foobar.yaml: no such file or directory`

But to me it feels that maybe this is the expected behaviour: Your included taskfile name needs to match the parent unless explicitly specified. 

An alternative would be to add a generic fallback for included dir which looks for foobar.yaml, Taskfile.yml, Taskfile.yaml (in that order, where first is the user specified taskfile name). But to me this feels a bit more "magical" and may create more confusion or require documentation.

Reply:
Thought about this a bit more and I think I'm now on-board with the behavior you described.